tests/cli/net_handler_cli.c

```diff
@@ -0,0 +1,424 @@
+#include "../client/net_handler_client.h"
+
+#define MAX_CMD_LEN 32
+
+// ********************************** COMMAND-SPECIFIC FUNCTIONS  ****************************** //
+
+void prompt_run_mode ()
+{
+	int client = SHEPHERD_CLIENT;
+	int mode = IDLE_MODE;
+	char nextcmd[MAX_CMD_LEN];
+	
+	//get client to send as
+	while (1) {
+		printf("Send as DAWN or SHEPHERD: ");
+		fgets(nextcmd, MAX_CMD_LEN, stdin);
+		if (strcmp(nextcmd, "dawn\n") == 0) {
+			client = DAWN_CLIENT;
+			break;
+		} else if (strcmp(nextcmd, "shepherd\n") == 0) {
+			client = SHEPHERD_CLIENT;
+			break;
+		} else if (strcmp(nextcmd, "abort\n") == 0) {
+			return;
+		} else {
+			printf("Invalid response to prompt: %s", nextcmd);
+		}
+	}
+	
+	//get mode to send
+	while (1) {
+		printf("Send mode IDLE, AUTO, or TELEOP: ");
+		fgets(nextcmd, MAX_CMD_LEN, stdin);
+		if (strcmp(nextcmd, "idle\n") == 0) {
+			mode = IDLE_MODE;
+			break;
+		} else if (strcmp(nextcmd, "auto\n") == 0) {
+			mode = AUTO_MODE;
+			break;
+		} else if (strcmp(nextcmd, "teleop\n") == 0) {
+			mode = TELEOP_MODE;
+			break;
+		} else if (strcmp(nextcmd, "abort\n") == 0) {
+			return;
+		} else {
+			printf("Invalid response to prompt: %s", nextcmd);
+		}
+	}
+	
+	//send
+	printf("Sending Run Mode message!\n\n");
+	send_run_mode(client, mode);
+}
+
+void prompt_start_pos ()
+{
+	int client = SHEPHERD_CLIENT;
+	int pos = LEFT_POS;
+	char nextcmd[MAX_CMD_LEN];
+	
+	//get client to send as
+	while (1) {
+		printf("Send as DAWN or SHEPHERD: ");
+		fgets(nextcmd, MAX_CMD_LEN, stdin);
+		if (strcmp(nextcmd, "dawn\n") == 0) {
+			client = DAWN_CLIENT;
+			break;
+		} else if (strcmp(nextcmd, "shepherd\n") == 0) {
+			client = SHEPHERD_CLIENT;
+			break;
+		} else if (strcmp(nextcmd, "abort\n") == 0) {
+			return;
+		} else {
+			printf("Invalid response to prompt: %s", nextcmd);
+		}
+	}
+	
+	//get pos to send
+	while (1) {
+		printf("Send pos LEFT or RIGHT: ");
+		fgets(nextcmd, MAX_CMD_LEN, stdin);
+		if (strcmp(nextcmd, "left\n") == 0) {
+			pos = LEFT_POS;
+			break;
+		} else if (strcmp(nextcmd, "right\n") == 0) {
+			pos = RIGHT_POS;
+			break;
+		} else if (strcmp(nextcmd, "abort\n") == 0) {
+			return;
+		} else {
+			printf("Invalid response to prompt: %s", nextcmd);
+		}
+	}
+	
+	//send
+	printf("Sending Start Pos message!\n\n");
+	send_start_pos(client, pos);
+}
+
+void prompt_gamepad_state ()
+{
+	uint32_t buttons = 0;
+	float joystick_vals[4];
+	char nextcmd[MAX_CMD_LEN];
+	char **list_of_names;
+	char button_name[16];
+	int button_ix = 0;
+	int set_joysticks = 0;
+	
+	//get which buttons are pressed
+	printf("Specify which of the following buttons are pressed:\n");
+	list_of_names = get_button_names();
+	for (int i = 0; i < NUM_GAMEPAD_BUTTONS; i++) {
+		printf("\t%s\n", list_of_names[i]);
```

Comment:
Right, so the CLI converts the button names to numbers before sending it to the client, it's easier for the CLI for people to enter the button names instead of the numbers. The test framework, however, uses the button numbers so that you don't have to do string manipulation in your tests.

Reply:
It's easier for people to enter numbers IMO. I don't want to type 10 characters each time, but maybe that's just me.

Reply:
Same as comment below